### ISSUE TITLE
Support opening STL & DXF files

### ISSUE THREAD

**Issue by jcoffland** [CauldronDevelopmentLLC/CAMotics#198]
This feature would open up a dialog asking the user for a handful of configuration options to turn a 2D STL or DXF in to a GCode program directly in CAMotics.

**Comment by martonmiklos** [comment]
Hi there!

Are you planning to integrate toolpath generation (CAM) to the software?
For dealing with DXF files I could recommend this library:
http://www.ribbonsoft.com/en/what-is-dxflib

It is written by teh QCAD guys.

**Comment by jcoffland** [comment]
Yes.  I did know about that library.  CAMotics already has a DXF lib.  You can see it here: https://github.com/CauldronDevelopmentLLC/CAMotics/tree/master/src/camotics/dxf  Thanks for the suggestion.

**Comment by martonmiklos** [comment]
Ah sorry, I have not checked the code. 
Do you have any plans to integrate toolpath generation to the software?

**Comment by jcoffland** [comment]
Yes, that's what this issue is about.

**Comment by jcoffland** [comment]
Initially what I have in mind for DXF import is very simple.  You open a .dxf file and are then presented with a wizard type dialog.  You are presented with a list of DXF layers which you may:

  * Inset cut
  * Outset cut
  * Cut on
  * Pocket
  * Island (i.e. polygons inside a pocket which should not be cut.

You can also set cutting the depth, step down, feed, speed, tool, etc.  When you hit OK it will generate the GCode and simulate.  This is very similar to the style of pyCAM.

This may not be the most advanced interface but it should cover a large number of common use cases and make CAMotics much more useful for many people.

**Comment by tpimh** [comment]
Any ideas on STL import?

**Comment by jcoffland** [comment]
It's not yet supported.
